spec/services/incoming_text_message_service_spec.rb

```diff
@@ -0,0 +1,119 @@
+        "SmsMessageSid" => "SM7067f0beef82c65f976dc2386a7sgd7w",
+        "NumMedia" => "0",
+        "ToCity" => "",
+        "FromZip" => "95050",
+        "SmsSid" => "SM7067f0beef82c65f976dc2386a7sgd7w",
+        "FromState" => "CA",
+        "SmsStatus" => "received",
+        "FromCity" => "LOS GATOS",
+        "Body" => body,
+        "FromCountry" => "US",
+        "To" => "+14158161286",
+        "ToZip" => "",
+        "NumSegments" => "1",
+        "MessageSid" => "SM7067f0beef82c65fb6785v46754v6754",
+        "AccountSid" => "AC70b4e3aa44fe961398q89we7yr98aw7y",
+        "From" => "+15005550006",
+        "ApiVersion" => "2010-04-01"
+    }
+  end
+
+  let(:current_time) { DateTime.new(2020, 9, 6) }
+  before do
+    allow(TwilioService).to receive(:valid_request?).and_return true
+    allow(DateTime).to receive(:now).and_return current_time
+    allow(ClientChannel).to receive(:broadcast_contact_record)
+  end
+
+  context "with a matching intake phone number" do
+    let(:client) { create :client }
+    let!(:intake) { create(:intake, client: client, phone_number: "+15005550006") }
+
+    it "creates a new IncomingTextMessage linked to the client the right data" do
+      expect do
+        subject
+      end.to change(IncomingTextMessage, :count).by 1
+
+      message = IncomingTextMessage.last
+      expect(message.body).to eq "Hello, it me"
+      expect(message.from_phone_number).to eq "+15005550006"
+      expect(message.received_at).to eq current_time
+      expect(message.client).to eq client
+    end
+
+    it "broadcasts the message" do
+      subject
+      expect(ClientChannel).to have_received(:broadcast_contact_record).with(IncomingTextMessage.last)
+    end
+  end
+
+  context "without a matching client" do
+    it "creates a new incoming text message attached to a new client" do
+      expect do
+        subject
+      end.to change(IncomingTextMessage, :count).by(1).and change(Client, :count).by(1)
+
+      message = IncomingTextMessage.last
+      expect(message.body).to eq "Hello, it me"
+      expect(message.from_phone_number).to eq "+15005550006"
+      expect(message.received_at).to eq current_time
+      client = Client.last
+      expect(message.client).to eq client
+      expect(client.intake.phone_number).to eq "+15005550006"
+      expect(client.intake.sms_phone_number).to eq "+15005550006"
+      expect(client.intake.sms_notification_opt_in).to eq("yes")
+      expect(client.vita_partner).to eq VitaPartner.client_support_org
+    end
+  end
+
+  context "with three matching client intakes" do
+    before do
+      create(:client, intake: (create :intake, phone_number: "+15005550006"))
+      create(:client, intake: (create :intake, phone_number: "+15005550006"))
+      create(:client, intake: (create :intake, sms_phone_number: "+15005550006"))
+      create(:client, intake: (create :intake, sms_phone_number: "+15005550005"))
+    end
+
+    it 'associates the messages with each existing client' do
```

Comment:
I suppose it'd be nice to validate the text message is on the client.

Maybe `.to change( { IncomingTextMessage.where(client: [client1, client2, client3]).count }.by(3)`